packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.tsx

```diff
@@ -53,22 +56,15 @@ const updateItemsState = ({
       );
     }
 
-    const label = getItemLabel ? getItemLabel(item) : (item as { label: string }).label;
-    if (label == null) {
-      throw new Error(
-        [
-          'MUI X: The Tree View component requires all items to have a `label` property.',
-          'Alternatively, you can use the `getItemLabel` prop to specify a custom label for each item.',
-          'An item was provided without label in the `items` prop:',
-          JSON.stringify(item),
-        ].join('\n'),
-      );
-    }
+    const editable = isItemEditable ? isItemEditable(item) : false;
+    const isBeingEdited = isItemBeingEdited ? isItemBeingEdited(id) && editable : false;
```

Comment:
Why do we need to store `isBeingEdited` here?
This forces `updateItemsState` to re-run when starting / stopping the edition.

If we access `isItemBeingEdited` through the `instance` method instead of a prop, it would simplify the state management no?

Reply:
By the way, I'm not even sure we need to store `editable` either, could we just call `isItemEditable` inside `instance.isItemBeingEdited` whenever we need to know if an item is being edited?

For me the best structure would be to have zero modification on `useTreeViewItems`, to keep the plugins independant.
It might not be doable, maybe we need some small adjustments on this plugin, but I think we can modify it a lot less :+1: